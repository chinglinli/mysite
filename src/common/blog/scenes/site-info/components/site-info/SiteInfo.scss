.siteInfo {
  float: left;
  width: 800px;

  @media (max-width: 1200px) {
    width: 100%;
  }

  article {
    min-height: 600px;

    padding: 2em;
    border-bottom: 1px solid #eee;
    background: white;

    header {
      h1 {
        margin: 3px 0;

        a {
          font-size: 2rem;
        }
      }
    }

    table {
      margin-top: 24px;

      border-collapse: collapse;
      width: 100%;
      font-size: 1.2em;

      tr {
        background: #eee;

        &:nth-of-type(even) {
          background: #f9f9f9;
        }

        td {
          padding: 0.6em;
          border: 1px solid #ccc;
        }
      }
    }
  }
}
