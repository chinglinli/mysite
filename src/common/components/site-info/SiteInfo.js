import * as React from 'react'

import styles from './SiteInfo.scss'

export default class SiteInfo extends React.Component {
  static PropTypes = {
    version: React.PropTypes.string.isRequired,
    updatedTime: React.PropTypes.string.isRequired
  }

  render () {
    return (
      <article className={styles.root}>
        <div>
          <header className={styles.header}>網站資訊</header>
          <table className={styles.table}>
            <tbody>
            <tr>
              <td>網站更新時間：</td>
              <td>{this.props.updatedTime}</td>
            </tr>
            <tr>
              <td>版本：</td>
              <td>{this.props.version}</td>
            </tr>
            </tbody>
          </table>
        </div>
      </article>
    )
  }
}