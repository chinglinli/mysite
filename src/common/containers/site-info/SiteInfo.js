import * as React from 'react'
import ImmutablePropTypes from 'react-immutable-proptypes'
import {connect} from 'react-redux'

import SiteHead from '../../components/site-head'
import * as siteSelectors from '../../ducks/site/selectors'
import * as configSelectors from '../../ducks/config/selectors'

import styles from './SiteInfo.scss'


export class SiteInfo extends React.Component {
  static PropTypes = {
    siteConfig: ImmutablePropTypes.map.isRequired,
    version: React.PropTypes.string.isRequired,
    updatedTime: React.PropTypes.string.isRequired
  }

  render() {
    return (
      <article className={styles.root}>
        <SiteHead config={this.props.siteConfig}/>
        <div>
          <header className={styles.header}>網站資訊</header>
          <table className={styles.table}>
            <tbody>
              <tr>
                <td>網站更新時間：</td><td>{this.props.updatedTime}</td>
              </tr>
              <tr>
                <td>版本：</td><td>{this.props.version}</td>
              </tr>
            </tbody>
          </table>
        </div>
      </article>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    siteConfig: siteSelectors.getSiteHeadConfig(state, ownProps),
    version: configSelectors.getSiteVersion(state),
    updatedTime: configSelectors.getSiteUpdatedTime(state)
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(SiteInfo)
