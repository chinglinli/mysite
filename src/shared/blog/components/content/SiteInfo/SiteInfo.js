import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import TitleLink from '../../generic/TitleLink'

const Base = styled.section`
  float: left;
  width: 800px;

  @media (max-width: 1200px) {
    width: 100%;
  }

  article {
    min-height: 600px;
    padding: 2em;
    border-bottom: 1px solid #eee;
    background: white;
  }
`

const Header = styled.header`
  h1 {
    margin: 3px 0 24px;

    a {
      font-size: 2rem;
    }
  }
`

const InfoTable = styled.table`
  border-collapse: collapse;
  width: 100%;
  font-size: 1.2em;

  tr {
    background: #eee;

    &:nth-of-type(even) {
      background: #f9f9f9;
    }

    td {
      padding: 0.6em;
      border: 1px solid #ccc;
    }
  }
`

export default class SiteInfo extends React.PureComponent {
  static PropTypes = {
    frontendVersion: PropTypes.string.isRequired,
    backendVersion: PropTypes.string.isRequired,
    updatedTime: PropTypes.string.isRequired
  }

  renderHeader = () => {
    return (
      <Header>
        <h1><TitleLink to='/info/'>網站資訊</TitleLink></h1>
      </Header>
    )
  }

  renderInfoTable = () => {
    return (
      <InfoTable>
        <tbody>
        <tr>
          <td>網站更新時間：</td>
          <td>{this.props.updatedTime}</td>
        </tr>
        <tr>
          <td>前端版本：</td>
          <td>{this.props.frontendVersion}</td>
        </tr>
        <tr>
          <td>後端版本：</td>
          <td>{this.props.backendVersion}</td>
        </tr>
        </tbody>
      </InfoTable>
    )
  }

  render () {
    return (
      <Base>
        <article>
          {this.renderHeader()}
          {this.renderInfoTable()}
        </article>
      </Base>
    )
  }
}