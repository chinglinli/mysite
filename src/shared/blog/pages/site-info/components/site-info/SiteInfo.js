import React from 'react'
import PropTypes from 'prop-types'
import Link from '../../../../../generic/components/Link'

import styles from './SiteInfo.scss'

export default class SiteInfo extends React.PureComponent {
  static PropTypes = {
    frontendVersion: PropTypes.string.isRequired,
    backendVersion: PropTypes.string.isRequired,
    updatedTime: PropTypes.string.isRequired
  }

  render () {
    return (
      <section className={styles.siteInfo}>
        <article>
          <header>
            <h1><Link to='/info/'>網站資訊</Link></h1>
          </header>
          <table>
            <tbody>
            <tr>
              <td>網站更新時間：</td>
              <td>{this.props.updatedTime}</td>
            </tr>
            <tr>
              <td>前端版本：</td>
              <td>{this.props.frontendVersion}</td>
            </tr>
            <tr>
              <td>後端版本：</td>
              <td>{this.props.backendVersion}</td>
            </tr>
            </tbody>
          </table>
        </article>
      </section>
    )
  }
}