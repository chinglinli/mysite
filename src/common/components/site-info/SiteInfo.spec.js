import Immutable from 'immutable'
import React from 'react'
import renderer from 'react-test-renderer'

import SiteInfo from './SiteInfo'


test('It should render SiteInfo component correctly', () => {
  const props = Immutable.fromJS({
    version: 'master (3f5eb304b501ea2b5437ed6c4612bbc66077b7b8)',
    updatedTime: '2017-04-08 12:01:57.198000'
  })

  const component = renderer.create(
    <SiteInfo
      version={props.get('version')}
      updatedTime={props.get('updatedTime')}
    />
  )
  expect(component.toJSON()).toMatchSnapshot()
})
