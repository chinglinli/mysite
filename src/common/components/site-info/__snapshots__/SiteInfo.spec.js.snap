// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render SiteInfo component correctly 1`] = `
<article
  className="root"
>
  <div>
    <header
      className="header"
    >
      網站資訊
    </header>
    <table
      className="table"
    >
      <tbody>
        <tr>
          <td>
            網站更新時間：
          </td>
          <td>
            2017-04-08 12:01:57.198000
          </td>
        </tr>
        <tr>
          <td>
            版本：
          </td>
          <td>
            master (3f5eb304b501ea2b5437ed6c4612bbc66077b7b8)
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</article>
`;
